hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/util/DelayedUtil.java

```diff
@@ -79,7 +84,13 @@ public String toString() {
    */
   public static <E extends Delayed> E takeWithoutInterrupt(final DelayQueue<E> queue) {
     try {
-      return queue.take();
+      E element = queue.poll(10, TimeUnit.SECONDS);
+      if (element == null && queue.size() > 0) {
+        LOG.error("DelayQueue is not empty when timed waiting elapsed. If this is repeated for"
```

Comment:
Since the default value of `hbase.procedure.remote.dispatcher.delay.msec` is just 150, I thought 10s might be enough. But I am open to keep it higher. What do you think would be better value?

Reply:
Also, this condition might be true in some case, but only if this condition continues to be `true` for the same queue consistently, we might have a problem. That's why I am keeping log message like this:
```
DelayQueue is not empty when timed waiting elapsed. If this is repeated for same queue consistently, it means no element is getting expired from DelayQueue and it might freeze the system.
```

Reply:
I do not think we should output a warn message if it is not a problem. As I said above, is it possible to move the warn log to upper layer? The DelayedUtil seems a general name, not only for dispatcher.